src/cpp/core/CMakeLists.txt

```diff
@@ -282,3 +281,35 @@ target_link_libraries(
    ${Boost_LIBRARIES}
    ${CORE_SYSTEM_LIBRARIES}
 )
+
+# define executable (for running unit tests)
+if (RSTUDIO_UNIT_TESTS_ENABLED)
+
+  file(GLOB_RECURSE CORE_TEST_FILES "*Tests.cpp")
+
+  add_executable(rstudio-core-bin
+     TestMain.cpp
+     ${CORE_TEST_FILES}
+     ${CORE_HEADER_FILES}
+   )
+
+   set_target_properties(rstudio-core-bin
+      PROPERTIES
+         OUTPUT_NAME rstudio-core
+   )
+
+   target_link_libraries(rstudio-core-bin
+      rstudio-core
+      rstudio-core-synctex
+      rstudio-core-hunspell
+      ${Boost_LIBRARIES}
+      ${CORE_SYSTEM_LIBRARIES}
+   )
+   
+   # run tests after build
```

Comment:
It might just be as simple as a convention that you always run the "smoke tests" before you check in. Alternatively we could have a post-commit hook somewhere else that builds and runs the tests.